src/commands/dev/dev.js

```diff
@@ -78,6 +79,28 @@ const isNonExistingCommandError = ({ command, error: commandError }) => {
   )
 }
 
+/**
+ * @type {(() => Promise<void>)[]} - array of functions to run before the process exits
+ */
+const cleanupWork = []
+
+let cleanupStarted = false
+
+/**
+ * @param {object} input
+ * @param {number=} input.exitCode The exit code to return when exiting the process after cleanup
+ */
+const cleanupBeforeExit = async ({ exitCode }) => {
+  if (cleanupStarted) {
+    // If cleanup has started, then wherever started it will be responsible for exiting
+  } else {
+    cleanupStarted = true
+    // eslint-disable-next-line no-unused-vars
+    const cleanupFinished = await Promise.all(cleanupWork.map((cleanup) => cleanup()))
+    process.exit(exitCode)
```

Comment:
Should this be in a `try`/`finally` code to ensure it is called even if one of the `cleanupWork` is rejected?

Reply:
Will do